flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java

```diff
@@ -137,6 +137,15 @@
                     .noDefaultValue()
                     .withDescription("JVM heap size (in megabytes) for the JobManager.");
 
+    /** The maximum stacktrace depth of JobManager's thread dump web-frontend displayed. */
+    @Documentation.Section(Documentation.Sections.ALL_JOB_MANAGER)
+    public static final ConfigOption<Integer> THREAD_DUMP_STACKTRACE_MAX_DEPTH =
+            key("jobmanager.thread-dump.stacktrace-max-depth")
```

Comment:
Do we want to have two configuration options or would a single one be easier for the user to control/configure?

Reply:
Two configuration options are more fine-grained but may introduce complexity.
Single one is easier. Because no need to specify JM and TM's thread dump depth, maybe single one is a better chioce.

Do you think so? If used single option, the config option should be written in which class? in `WebOptions` ?

Reply:
We could put them in `ClusterOptions` maybe.

Reply:
Updated. Please recheck. Thanks @tillrohrmann